File: example1.php
<?php
 
// Пример на PHP
 
class User {
   public $id;
   public $name;
   public $age;
 
   public function __construct($id, $name, $age) {
       $this->id = $id;
       $this->name = $name;
       $this->age = $age;
   }
}
 
function printUsers($users)
{
   for ($i = 0; $i < count($users); $i++) {
       echo "users[" . $i . "] = { ID: " . $users[$i]->id .
            ", name: " . $users[$i]->name .
            ", age: " . $users[$i]->age . " }" . PHP_EOL;
   }
}
 
function sortUsers(&$users)
{
   $count = count($users);
   $n = $count - 1;

   for ($i = 0; $i < $count; $i++) {
      $changes = false;
      for ($j = 0; $j < $n; $j++) {
         if ($users[$j] > $users[($j + 1)]) {
            $changes = true;
            list($users[$j], $users[($j + 1)]) = array($users[($j + 1)], $users[$j]);
         }
      }
      $n--;
      if (!$changes) {
         return $users;
      }
   }
   return $users;
}

 
 $users = [
   new User(10, "Игорь", 23),
   new User(3, "Вася", 17),
   new User(1, "Елена", 18),
   new User(9, "Ольга", 20),
   new User(7, "Виктор", 33),
   new User(6, "Лариса", 21),
   new User(4, "Гена", 48),
   new User(2, "Олег", 24),
   new User(5, "Женя", 32),
   new User(8, "Кристина", 13)
 ];
 
 
 sortUsers($users);
 
 printUsers($users);

File: example1.cpp
#include <iostream>
#include <string>

using namespace std;

struct User
{
int ID;
string name;
int age;
};

// Семантика функции, которую нужно реализовать
// users - массив пользователей
// N - размер массива
void sort(User *users, int N)
{
    for (int i = 0; i < N; i++)
    {
        for (int k = i; k < N; k++)
        {
            if (users[i].ID > users[k].ID)
            {
                swap(users[i],users[k]);
            }
        }
    }
}

//Функция перестановки местами значений
void swap(User x, User y)
{
    User temp = x;
    x = y;
    y = temp;
}

void print(User *users, int N)
{
    for (int i = 0; i < N; i++)
    {
        cout << "users[" << i << "] = { ID: " << users[i].ID << ",  name: " << users[i].name << ", age = " << users[i].age << " }" << endl;
    }
}

int main()
{
    User users[10] = {
        {10, "Игорь", 23},
        {3, "Вася", 17},
        {1, "Елена", 18},
        {9, "Ольга", 20},
        {7, "Выктор", 33},
        {6, "Лариса", 21},
        {4, "Гена", 48},
        {2, "Олег", 24},
        {5, "Женя", 32},
        {8, "Кристина", 13}};

    sort(users, 10);

    print(users, 10);

    return 0;
}
